plugins/jenkins-backend/src/service/jenkinsApi.ts

```diff
@@ -318,4 +322,45 @@ export class JenkinsApiImpl {
       })
       .pop();
   }
+
+  private getBuildUrl(
+    jenkinsInfo: JenkinsInfo,
+    jobFullName: string,
+    buildId: number,
+  ): string {
+    const jobs = jobFullName.split('/');
+    return `${jenkinsInfo.baseUrl}/job/${jobs.join('/job/')}/${buildId}`;
+  }
+
+  private async getHeaders(jenkinsInfo: JenkinsInfo): Promise<HeaderInit> {
```

Comment:
you don't need a crumb issuer, you should be using an API token not a username and password 

https://www.jenkins.io/doc/book/security/csrf-protection/#working-with-scripted-clients

> Alternatively authenticate using username and API token. Requests authenticating with an [API token](https://www.jenkins.io/doc/book/system-administration/authenticating-scripted-clients/) are exempt from CSRF protection in Jenkins.

Reply:
That's what I thought too. However, there was an existing check that explicitly checked for it. I'm happy to remove it if you don't think it's needed.

Reply:
existing check where?

Reply:
added in https://github.com/backstage/backstage/pull/8500

I wouldn't have recommended it.
There's no reason in a scripted client to not use API tokens as far as I know.

unfortunately PR gives no reasoning or issue to why it was being added.

my recommendation would be to drop it =/, but I guess this is okay too if you want to maintain compatibility.

Reply:
I agree that it should be dropped if not needed. Thank you so much for your help improving this PR!